src/utils/proofs_api/paramfetch.rs

```diff
@@ -14,28 +18,28 @@ use crate::{
     shim::sector::SectorSize,
     utils::net::{download_ipfs_file_trustlessly, global_http_client},
 };
-use ahash::HashMap;
 use anyhow::{bail, Context};
 use backoff::{future::retry, ExponentialBackoffBuilder};
-use blake2b_simd::{Hash, State as Blake2b};
-use cid::Cid;
-use futures::{AsyncWriteExt, TryStreamExt};
-use serde::{Deserialize, Serialize};
+use futures::{stream::FuturesUnordered, AsyncWriteExt, TryStreamExt};
 use tokio::fs::{self};
-use tracing::{debug, error, info, warn};
+use tracing::{debug, info, warn};
+
+use super::{
+    is_env_truthy,
+    parameters::{
+        check_parameter_file, param_dir, ParameterData, ParameterMap, DEFAULT_PARAMETERS,
+        PROOFS_PARAMETER_CACHE_ENV,
+    },
+};
 
-const GATEWAY: &str = "https://proofs.filecoin.io/ipfs/";
-const PARAM_DIR: &str = "filecoin-proof-parameters";
-const DEFAULT_PARAMETERS: &str = include_str!("./parameters.json");
+const IPFS_GATEWAY: &str = "https://proofs.filecoin.io/ipfs/";
 
 /// Domain bound to the Cloudflare R2 bucket.
 const CLOUDFLARE_PROOF_PARAMETER_DOMAIN: &str = "filecoin-proof-parameters.chainsafe.dev";
 
 /// If set to 1, enforce using the IPFS gateway for fetching parameters.
 const PROOFS_ONLY_IPFS_GATEWAY_ENV: &str = "FOREST_PROOFS_ONLY_IPFS_GATEWAY";
-const DIR_ENV: &str = "FIL_PROOFS_PARAMETER_CACHE";
-const GATEWAY_ENV: &str = "IPFS_GATEWAY";
-const TRUST_PARAMS_ENV: &str = "TRUST_PARAMS";
+const IPFS_GATEWAY_ENV: &str = "IPFS_GATEWAY";
```

Comment:
Maybe worth mentioning in doc somewhere
https://lotus.filecoin.io/kb/nodes-in-china/
```
Running Lotus requires the download of chain’s proof parameters which are large files which by default are hosted outside of China and very slow to download there. To get around that, users should set the following environment variable when running either of lotus, lotus-miner and lotus-worker:

export IPFS_GATEWAY=https://proof-parameters.s3.cn-south-1.jdcloud-oss.com/ipfs/
```

Reply:
At first glance I feel we should prefix the environment variable with `FOREST_` but in this case maybe it's more convenient to re-use the lotus one. They don't prefix it with `LOTUS_` for some reason.

Reply:
> At first glance I feel we should prefix the environment variable with `FOREST_` but in this case maybe it's more convenient to re-use the lotus one. They don't prefix it with `LOTUS_` for some reason.

Agreed on the prefix. As stated above - it would be nice to have some sort of tool that would allow us to specify the prefix for everything we load. Then we can load variables with and without the prefix.

Reply:
@hanabi1224 Great point there, I was not aware of that. On top of that, I was able to find a bug (not sure how long it was there) where the `IPFS_GATEWAY` was not overridable - the environment variable only changed what was logged. :clown_face: See [a4c53c5](https://github.com/ChainSafe/forest/pull/4177/commits/a4c53c53ea7ff6cf0115c24539b2efecc6eb3661)